test/token.test.ts

```diff
@@ -164,62 +158,152 @@ describe('token integration', () => {
       await tx.prove();
       await tx.send();
     });
+  });
 
-    it('should deploy a third party token account for token A', async () => {
 
-      const tx = await Mina.transaction(context.deployerAccount, () => {
-        context.tokenAccountA.deploy({ ownerAddress: context.tokenAAccount });
-        context.tokenA.approve(context.tokenAccountA.self);
-      });
+  describe('mint/burn', () => {
+    const mintAmount = UInt64.from(1000);
+    const burnAmount = UInt64.from(100);
 
-      tx.sign([context.deployerKey, context.thirdPartyKey]);
+    it('should mint for the sender account', async () => {
+      const initialBalance = context.tokenA.getBalanceOf(context.senderAccount)
+        .toBigInt()
 
+      const tx = await Mina.transaction(context.senderAccount, () => {
+        AccountUpdate.fundNewAccount(context.senderAccount, 2);
+        context.tokenA.mint(context.senderAccount, mintAmount);
+      });
+  
+      tx.sign([context.senderKey, context.tokenAdminKey]);
```

Comment:
still wondering why we enforse an admin to sign a burn transaction.
I can't imagine this to be ever applied in production (signing both from user and admin)

In the reference implementation , we should enforce tx to be signed either by user or by an admin (I suggest user)

Reply:
Ok, I'll remove the requirement for the admin signature, and replace it with a comment that you could require it if you so desired.